src/api/search/test/query.test.js

```diff
@@ -1,9 +1,230 @@
+      expect(res.status).toBe(400);
+      expect(res.body[0].value).toBe('1.5');
+
+      res = await request(app).get('/advanced').query({ post: 'ElasticSearch', perPage: 'one' });
+      expect(res.status).toBe(400);
+      expect(res.body).toStrictEqual([
+        {
+          value: 'one',
+          msg: 'perPage should be empty or a number between 1 to 10',
+          param: 'perPage',
+          location: 'query',
+        },
+      ]);
+    });
+
+    it('Return error 400 if "page" param is not an integer between 0 to 999', async () => {
+      let res = await request(app).get('/advanced').query({ post: 'ElasticSearch', page: 1101 });
+      expect(res.status).toBe(400);
+      expect(res.body[0].value).toBe('1101');
+
+      res = await request(app).get('/advanced').query({ post: 'ElasticSearch', page: -1 });
+      expect(res.status).toBe(400);
+      expect(res.body[0].value).toBe('-1');
+
+      res = await request(app).get('/advanced').query({ post: 'ElasticSearch', page: 55.5 });
+      expect(res.status).toBe(400);
+      expect(res.body[0].value).toBe('55.5');
+
+      res = await request(app).get('/advanced').query({ post: 'ElasticSearch', page: 'five' });
+      expect(res.status).toBe(400);
+      expect(res.body).toStrictEqual([
+        {
+          value: 'five',
+          msg: 'page should be empty or a number between 0 to 999',
+          param: 'page',
+          location: 'query',
+        },
+      ]);
+    });
+  });
+
+  describe('Test queries that pass validation', () => {
+    const { mock } = Elastic();
+
+    afterEach(() => {
+      mock.clearAll();
+    });
+
+    it('Return status 200 if params pass validation checks', async () => {
+      const mockResults = {
+        hits: {
+          total: { value: 0 },
+          hits: [],
+        },
+      };
+
+      mock.add(
+        {
+          method: ['POST', 'GET'],
+          path: '/posts/post/_search',
+        },
+        () => {
+          return mockResults;
+        }
+      );
+
+      let res = await request(app).get('/advanced').query({ post: 'ElasticSearch' });
+      expect(res.status).toBe(200);
+
+      res = await request(app).get('/advanced').query({ author: 'David' });
+      expect(res.status).toBe(200);
+
+      res = await request(app).get('/advanced').query({ title: 'OSD' });
+      expect(res.status).toBe(200);
+
+      res = await request(app).get('/advanced').query({
+        post: 'ElasticSearch',
+        author: 'David',
+        title: 'OSD',
+        from: '2020-04-06',
+        to: '2019-06-02',
+        perPage: '2',
+        page: '1',
+      });
+      expect(res.status).toBe(200);
+      expect(res.body).toStrictEqual({ results: 0, values: [] });
+    });
+
+    it('Search results of more than 0 return "id" and "url" of posts', async () => {
+      const mockResults = {
+        hits: {
+          total: { value: 2 },
+          hits: [{ _id: '1111' }, { _id: '2222' }],
+        },
+      };
```

Comment:
Can you explain to me the properties of `mockResult`?

Reply:
so `hits` is the total number of result returned from the query and the array of post's id?

Reply:
Yes. As the [documentation ](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-search.html#search-api-response-body)says. The outer `hits` object contains the returned documents and metadata. The inner `hits` is an array of returned documented objects.